core/variant.cpp

```diff
@@ -601,6 +601,31 @@ bool Variant::can_convert_strict(Variant::Type p_type_from, Variant::Type p_type
 	return false;
 }
 
+bool Variant::can_convert(Type p_type_to) const {
+	// If this is a string variant and we are converting to an integer, check if the contents of the string
+	// make sense as an integer
+	if (type == Type::STRING && p_type_to == Type::INT) {
```

Comment:
This can be expanded to also include floats and booleans, since those can also come from strings.

I'm not entirely sure how to use regex without ``std::regex`` and ``std::regex_match``, but ``std::regex pattern(R"((-?\d+)(\.\d+)?|true|false)");`` would cover most of the bases, minus the String being either a 1 or a 0 when converting to a Bool.

It'll also catch cases where the String can be "-1a", which your code would catch as a valid Int, even though it's not.

Reply:
I'll look into it and find a solution to this issue.

Reply:
Using the pattern `^((-?\\d+)(\\.\\d+)?|true|false)$` in Godot will correctly match positive or negative numbers with or without a decimal part as well as `true` or `false`. So it should be possible to use Godot's own regular expression engine to perform a better test on the string. However, the regular expression will need to be compiled once at startup for performance reasons. I believe I can find a way to implement this though.

Reply:
I appears that having the regular expression be global causes a memory leak in the Linux template builds. I'll have to try making it a local variable instead.

Reply:
Making the regular expression local fixed the memory leak issue, but the minimal template build for Linux is failing due to a link error. The cause appears to be that the `regex` module does not get built for that particular build and therefore cannot be used by `Variant`. The only viable option without using the `regex` module would be to use `std::regex`. However, that may not be desirable for this project. I'm open to suggestions on how to proceed with this.